developing-packages.md

```diff
@@ -0,0 +1,175 @@
+---
+layout: page
+title: Developing Packages & Plugins
+permalink: /developing-packages/
+---
+
+* TOC
+{:toc}
+
+## Package introduction
+
+Dart packages enable the creation of modular code that can easily be shared. A
+minimal package with a single library consists of:
+
+* A `pubspec.yaml` file: A metadata file that declared the package name,
+  version, author, etc.
+
+* A `lib` directory containing the public code in the package, minimally a
+  single `<package-name>.dart` file.
+
+For additional details, see the [Dart library package](https://www.dartlang.org/guides/libraries/create-library-packages) documentation.
+
+### Package types {#types}
+
+Flutter supports several kinds of packages:
```

Comment:
I don't see the confusion; actually, I find what you are suggesting more confusing, and over-abstracted.

To use an analogy, the term "xml file" is commonly used rather than the more generic "text file containing text in xml format". It's just convenient to use more concrete terms.